From 8a52b9d3dae64d41d07601220988c136065854e4 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 21 Oct 2020 13:02:22 +0200
Subject: [PATCH 63/N] fixup! Don't create WSL symlinks for MSYS2

Cygwin started to use WSL-style symbolic links by default.

Since MSYS2 decided to patch the `WSYM_sysfile` mode to make deep copies
instead, that logic had to be disabled specifically.

But now that we introduced a separate mode (`WSYM_deepcopy`) for that,
and use that by default, we can re-enable Cygwin's code so that users
wishing to experience Cygwin's symbolic link behavior can do so by
setting `MSYS=winsymlinks:sysfile`.

Essentially, this drops the patch "Don't create WSL symlinks for MSYS2".

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index f780de5..4813bba 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -1997,7 +1997,6 @@ typedef struct _REPARSE_LX_SYMLINK_BUFFER
   } LxSymlinkReparseBuffer;
 } REPARSE_LX_SYMLINK_BUFFER,*PREPARSE_LX_SYMLINK_BUFFER;
 
-#ifndef __MSYS__
 static int
 symlink_wsl (const char *oldpath, path_conv &win32_newpath)
 {
@@ -2068,7 +2067,6 @@ symlink_wsl (const char *oldpath, path_conv &win32_newpath)
   NtClose (fh);
   return 0;
 }
-#endif
 
 int
 symlink_worker (const char *oldpath, path_conv &win32_newpath, bool isdevice)
@@ -2153,7 +2151,6 @@ symlink_worker (const char *oldpath, path_conv &win32_newpath, bool isdevice)
 	    }
 	  /* Otherwise, fall back to default symlink type. */
 	  wsym_type = WSYM_sysfile;
-#ifndef __MSYS__
 	  fallthrough;
 	case WSYM_sysfile:
 	  if (win32_newpath.fs_flags () & FILE_SUPPORTS_REPARSE_POINTS)
@@ -2165,7 +2162,6 @@ symlink_worker (const char *oldpath, path_conv &win32_newpath, bool isdevice)
 	  /* On FSes not supporting reparse points, or in case of an error
 	     creating the WSL symlink, fall back to creating the plain old
 	     SYSTEM file symlink. */
-#endif
 	  break;
 	default:
 	  break;
-- 
2.9.0

